### PR TITLE
graph: Set roles/actions in `sharedByMe` response

### PR DESCRIPTION
**Pull request by rhafer**
Try to map CS3 resource permissions on a share to one of the default libregraph
UnifiedRoleDefinitions. If a match if found return the roleid in 'permissions.roles'
attribute of the response. If no match if found convert the
ResourcePermissions in to `libre.graph.permissions.actions` and return
those in the response.

### CODE REVIEW

File: services/graph/pkg/service/v0/sharedbyme.go [hunk 1]
```diff
@@ -160,7 +161,10 @@ func (g Graph) cs3UserSharesToDriveItems(ctx context.Context, shares []*collabor
 		if s.GetExpiration() != nil {
 			perm.SetExpirationDateTime(cs3TimestampToTime(s.GetExpiration()))
 		}
-
+		actions := unifiedrole.CS3ResourcePermissionsToLibregraphActions(*s.GetPermissions().GetPermissions())
```
Comment (rhafer) [inline services/graph/pkg/service/v0/sharedbyme.go:164]: Note to self: only set `action` if a matching role can't be found.

**Comment by rhafer** [comment]
There is still a issue that cause the `Editor` role, when set via `ocs` on a file (not on a folder) to not be correctly mapped back to the `FileEditor` unifiedrole. Some part of the fix needs to happen in reva (I am  currently working on that). I don't think we need to hold this back for the fix though.

File: services/graph/pkg/unifiedrole/unifiedrole.go [hunk 2]
```diff
@@ -0,0 +1,346 @@
+package unifiedrole
```
Comment (2403905) [inline services/graph/pkg/unifiedrole/unifiedrole.go:1]: What reason to copy the mapping from a reva?

Comment (rhafer) [reply services/graph/pkg/unifiedrole/unifiedrole.go:1]: We're basically moving it here. See: https://github.com/cs3org/reva/pull/4336

**Review by 2403905 — APPROVED** [review]

**Comment by rhafer** [comment]
rebased

**Review by fschade — APPROVED** [review]
🎸

**Pull request merged**